pkg/ebpf/precompiled/deprecate.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package precompiled implements precompiled specific eBPF functionality
+package precompiled
```

Comment:
I realize the config uses the term `precompiled`, but we use the term prebuilt pretty much everywhere else. I'd like to be consistent and use `prebuilt` where possible.

Reply:
Even in the log messages? I have no preference either way.

Reply:
Yeah. WDYT about deprecating the `precompiled` config value and adding one using `prebuilt`?

Reply:
We would have to still get rid of the `precompiled` key default for the deprecation to work correctly I think, but I need to verify this.

Reply:
Correct, you cannot have a default if you need to detect if the value is set or not.

Reply:
Done, please verify that the new test in `config_linux_test.go` does the right thing.